### PR TITLE
feat(testing): update to jest v24

### LINKED ISSUE #1029: Jest 24 and jest-preset-angular 7 changes
After updating to latest Jest(24.0.0) and jest-angular-preset(7.0.0-alpha.2) which has some breaking changes:
https://github.com/thymikee/jest-preset-angular/blob/release/7/CHANGELOG.md

This is only showing warnings for now but  the main problem is that it does not find `tsconfig.spec.json` anymore.

Debugging `jest-builder` showed that change in https://github.com/nrwl/nx/blob/7.1.x/packages/builders/src/jest/jest.builder.ts#L47:
```
tsConfig: path.join('<rootDir>', path.relative(builderConfig.root, options.tsConfig))
```
fixes tests.

Also its weird that I cannot override `globals` Jest settings in my `jest.config.js` which I think would be expected.

# Prerequisites
- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed
- [X] I'm reporting the issue to the correct repository (not related to Angular, AngularCLI or any dependency)

## Expected Behavior
Running jest unit tests  normally as was before an update to latest Jest and jest-preset-angular

## Current Behavior
```
File not found: tsconfig.spec.json (resolved as: .../tsconfig.spec.json)
```

It resolves always to monorepo root folder, not to lib folder.

### PR DESCRIPTION
**Pull request by FrozenPandaz**
## Current Behavior (This is the behavior we have today, before the PR is merged)

* Jest is at v23
* `jest-angular-preset` is installed for projects without angular

## Expected Behavior (This is the new behavior we can expect after the PR is merged)

* Jest is at v24
* `jest-angular-preset` is not installed for projects without angular, `ts-jest` is

## Issue
Fixes #1029

### CODE REVIEW

**Comment by FrozenPandaz** [comment]
Wait until `7.0.0` final is released of `jest-angular-preset`

**Comment by ViceIce** [comment]
Any chance to get a preview release?

**Comment by ViceIce** [comment]
[jest-preset-angular](https://www.npmjs.com/package/jest-preset-angular) v7 is finally released.

File: packages/builders/plugins/jest/resolver.ts [hunk 1]
```diff
@@ -37,6 +37,16 @@ function getCompilerSetup(rootDir: string) {
...
 module.exports = function(path: string, options: ResolveOptions) {
+  const ext = extname(path);
+  if (
+    ext === '.css' ||
+    ext === '.scss' ||
+    ext === '.sass' ||
+    ext === '.less' ||
+    ext === '.styl'
+  ) {
+    return require.resolve('identity-obj-proxy');
```
Comment (vsavkin) [inline packages/builders/plugins/jest/resolver.ts]: why is this necessary?

**Review by vsavkin — APPROVED** [review]

File: packages/builders/plugins/jest/resolver.ts (hunk 1 above)
Comment (FrozenPandaz) [reply packages/builders/plugins/jest/resolver.ts]: react apps expect styles to be an object when imported? Not totally sure.. @jaysoo has more info.
